src/rosdistro/manifest_provider/github.py

```diff
@@ -39,32 +39,24 @@
     from urllib2 import URLError
 
 from rosdistro import logger
-from rosdistro.manifest_provider import get_release_tag
-from rosdistro.manifest_provider.git import check_remote_tag_exists
 
 
 def github_manifest_provider(_dist_name, repo, pkg_name):
     assert repo.version
-    if 'github.com' not in repo.url:
+    server, path = repo.get_url_parts()
+    if server != 'github.com':
         logger.debug('Skip non-github url "%s"' % repo.url)
         raise RuntimeError('can not handle non github urls')
 
-    release_tag = get_release_tag(repo, pkg_name)
+    release_tag = repo.get_release_tag(pkg_name)
 
-    if not check_remote_tag_exists(repo.url, release_tag):
+    if release_tag not in repo.remote_tags:
```

Comment:
By changing the way of the condition ("does tag x exist" vs. "x in list-of-tags") this limits future implementations since the API now always needs to get all remote tags. Can we change this to be a method on `RepositorySpecification` but with the old semantic (`has_remote_tag(tag_name)`)?